src/test/java/uk/gov/register/presentation/resource/HistoryResourceTest.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.register.presentation.resource;
+
+import org.junit.Test;
+
+import javax.ws.rs.NotFoundException;
+
+import static org.junit.Assert.fail;
+
+public class HistoryResourceTest {
+    @Test
+    public void history_throwsNotFoundExceptionForNonPrimaryKeyRequests() {
+        RequestContext requestContext = new RequestContext() {
+            @Override
+            public String getRegisterPrimaryKey() {
+                return "localhost";
+            }
+        };
+
+        HistoryResource resource = new HistoryResource(requestContext, null);
+
+        try {
+            resource.history("someOtherKey", "value");
+            fail("Must fail");
+        } catch (NotFoundException e) {
```

Comment:
you can use `@Test(expected = NotFoundException.class)` to avoid the try/catch block here